.github/workflows/mypy.yml

```diff
@@ -0,0 +1,42 @@
+name: mypy
+
+on:
+  push:
+    paths:
+      - '**.py'
+  pull_request:
+    paths:
+      - '**.py'
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # https://github.com/actions/checkout/issues/249
+          fetch-depth: 0
+          submodules: true
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+          cache: pip
+          cache-dependency-path: |
+            **/pyproject.toml
+            **/requirements*.txt
+
+      - name: Remove llama-cpp-python from cache
+        run: |
+          pip cache remove llama_cpp_python
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox
+
+      - name: Run mypy check
+        run: |
+          tox -e mypy
```

Comment:
I would rather see this added as another step in https://github.com/instructlab/instructlab/blob/main/.github/workflows/lint.yml instead of a new workflow.

Reply:
It's not hard to do that of course. But wouldn't we want to see both results at the same time and not wait for basic lint checks to pass before seeing `mypy` fail? Is there a drawback in having these separated? (Legit question, I am new to building github workflows.)

Reply:
It is maintenance of more workflows that would be good to avoid. Since the other workflow already sets up the environment (pip install, etc), it would be good to avoid. And linting is not the long pole in a PR build. It is the test workflow.

Reply:
> not the long pole in a PR build

I was thinking more about giving an author / reviewer immediate feedback on both linters and type-checks (instead of waiting for a new push to reveal type-checks failing further down the manifest.) But I will oblige here. (I wonder if maintenance cost could be reduced by creating common meta-actions for all workflows...)

Reply:
> instead of waiting for a new push to reveal type-checks failing further down the manifest

This could be handled by something like:

```yaml
if: ${{ always() && ((steps.lint.outcome == 'success') || (steps.lint.outcome == 'failure')) }}

```

on the mypy step so that it runs even if the lint step fails. So you would get both check in a single PR run.

Reply:
I made it depend on tox installation step success. I think this is what we want - to have further checkers execute independently of each other, as long as tox is installed.